mountpoint-s3-client/src/endpoint.rs

```diff
@@ -117,8 +126,10 @@ pub enum EndpointError {
     InvalidUri(#[from] InvalidUriError),
     #[error("endpoint URI cannot include path or query string")]
     InvalidEndpoint,
-    #[error("region {0} is not yet supported")]
+    #[error("region {0} could not be resolved")]
     UnsupportedRegion(String),
+    #[error("Endpoint could not be resolved")]
+    UnresolvedEndpoint(#[from] mountpoint_s3_crt::common::error::Error),
```

Comment:
This should also include the error string we got back from the resolver, which will explain _why_ we couldn't resolve an endpoint.